src/server/test/db/unitReadingsTests.js

```diff
@@ -0,0 +1,605 @@
+			const start = moment(startDate);
+			const end = moment(startDate).clone();
+			const numDays = 61;
+			end.add(numDays, 'days');
+			const allReadings = await Reading.getMeterLineReadings([meter.id], graphicUnitId, start, end, conn);
+			// The min days is 61 for daily so should get daily points so 61.
+			expect(allReadings[meter.id].length).to.equal(numDays);
+		})
+
+		mocha.it('Use Daily resolution when 60 days', async () => {
+			const start = moment(startDate);
+			const end = moment(startDate).clone();
+			const numDays = 60;
+			end.add(numDays, 'days');
+			const allReadings = await Reading.getMeterLineReadings([meter.id], graphicUnitId, start, end, conn);
+			// The min days is 61 for daily so should get numDays days of hourly readings.
+			expect(allReadings[meter.id].length).to.equal(numDays * 24);
+		});
+
+		mocha.it('Us raw readings when 13 days', async () => {
+			const start = moment(startDate);
+			const end = moment(startDate).clone();
+			const numDays = 13;
+			end.add(numDays, 'days');
+			const allReadings = await Reading.getMeterLineReadings([meter.id], graphicUnitId, start, end, conn);
+			// The min days is 14 for hourly so should get numDays days of 15-minute readings.
+			// This is four since there are four 15-minute intervals in one hour.
+			const hourToRawScaleFactor = 4;
+			expect(allReadings[meter.id].length).to.equal(numDays * 24 * hourToRawScaleFactor);
+		});
+	});
+
+	// TODO modify so checks values too.
+
+	mocha.describe('Number of readings for underlying 23-minute data', () => {
+		let meter, graphicUnitId, conn;
+		const startDate = '2020-01-01 00:00:00';
+		// Do 61 days which is what is needed.
+		const endDate = '2020-03-02 00:00:00';
+
+		mocha.beforeEach(async function () {
+			// Extend timeout because a longer time with more data being created. The value is somewhat
+			// arbitrary and can be made larger if you get timeouts.
+			this.timeout(20000);
+			conn = testDB.getConnection();
+			// Insert the standard and special units and conversions. Really only need 1-2 but this is easy.
+			await Unit.insertStandardUnits(conn);
+			await Conversion.insertStandardConversions(conn);
+			await insertSpecialUnits(conn);
+			await insertSpecialConversions(conn);
+			await redoCik(conn);
+			// Make the meter be a kWh meter.
+			const meterUnitId = (await Unit.getByName('Electric_utility', conn)).id;
+			await new Meter(undefined, 'Meter', null, false, true, Meter.type.OTHER, 'CST', undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined,
+				undefined, undefined, undefined, undefined, meterUnitId, meterUnitId).insert(conn);
+			meter = await Meter.getByName('Meter', conn);
+			// Make the graphic unit be MegaJoules.
+			graphicUnitId = (await Unit.getByName('MJ', conn)).id;
+			const data = generateSineData(startDate, endDate, { timeStep: { minute: 15 } }).map(row => new Reading(meter.id, row[0], row[1], row[2]));
```

Comment:
This is the test for underlying 23-minute data but the timeStep is set as 15. Do you think we should change it to 23?

Reply:
Oops. I changed it. I had to update the point it crossed per the formula and pushed the 15 min. one to the absolute limit. I clarified the comments and formulas to be consistent for 15 and 23 minute tests. I also had to change the reading rate for 23 min. so it really calculated for 23 minutes. After that the tests seem to all pass. Thanks.